### PR TITLE
Update for oss-fuzz Issue 22106

### PR DESCRIPTION
**Pull request by twghu**
Signed-off-by: Tim Walsh <temporal.differential@gmail.com>

[envoy:codec_impl_fuzz_test: Crash in Envoy::Http::HttpStream::directionalAction](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=22106&q=label%3AReproducible%20envoy&can=2)

Commit Message: Update for oss-fuzz Issue 22106
Additional Description: Note that [#4709](https://github.com/envoyproxy/envoy/issues/4709) covers a number of [oss-fuzz error](https://bugs.chromium.org/p/oss-fuzz/issues/list?q=label%3AReproducible%20envoy&can=2) This is part of the ongoing effort started for #4709.
Risk Level: Low
Testing: test/common/http
Docs Changes: None
Release Notes: 
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/common/stats/stat_merger_fuzz_test.cc
Comment (antoniovicente) [inline test/common/stats/stat_merger_fuzz_test.cc:43]: There seems to be an inconsistency between the line above and the comment before it.  The code above generate num_bytes between 0 and 7, not 1 and 8.

https://github.com/envoyproxy/envoy/pull/10965/files#r629769284

File: test/common/stats/stat_merger_fuzz_test.cc
Comment (antoniovicente) [inline test/common/stats/stat_merger_fuzz_test.cc:44]: is `num_bytes == 0` ok if `index != 0` ?

File: test/common/stats/stat_merger_fuzz_test.cc
Comment (jmarantz) [reply test/common/stats/stat_merger_fuzz_test.cc:43]: I think you are right; the code should be changed to reflect the comment. However it's not material to the problems in this fuzz test.

There's an outstanding PR that will fix it, however: https://github.com/envoyproxy/envoy/pull/16239

@jessicayuen will you be able to push that forward?

I think the fix offered here is too aggressive; it doesn't just skip over the troublesome cases but completely stops the test on the passed-in corpus. It could just 'continue'.

File: test/common/stats/stat_merger_fuzz_test.cc
Comment (twghu) [reply test/common/stats/stat_merger_fuzz_test.cc]: [Change to initial byte count](https://github.com/twghu/envoy/commit/3f3e6848c18b5664d923446135c409f7c2898bac) has resolved crash and removed need for additional testing of num_bytes.

**Review by jmarantz — APPROVED** [review]
lgtm from my perspective. Antonio do you want to take another look?

**Comment by jmarantz** [comment]
I don't know why CI is not completing on this PR.  @envoyproxy/maintainers any ideas?

**Comment by asraa** [comment]
FWIW when it hangs like this I haven't found any way to solve but an extra commit (e.g. merge)

Thanks so much for this PR btw!

**Comment by jmarantz** [comment]
@twghu can you 'merge main' and then push? Hopefully that will kick CI and we can then merge this. Thank you!

/wait

**Comment by phlax** [comment]
@twgu - not sure why this happens sometimes with merging - it can be fixed with a rebase and force-push

**Comment by twghu** [comment]
Uggh This has got too messy. 
I merged main branch and pushed
then I attempted to rebase "work"  branch.

The only commits needed (can be picked?)  are the latest I added 22/5 

Maybe a new PR is best option and close this one.

**Comment by phlax** [comment]
up to you - but what will work assuming you have main envoy set as `upstream` remote

```
git fetch upstream
git rebase upstream/main
git push -f
```

you also might need to fix dco - but that could be an artefact of merge mash

**Comment by cpakulski** [comment]
Yeah, I encountered the same situation in the past and closed the PR and opened a new one.

To go a round missing DCOs I push my changes first without merging upstream/main. It executes all checks.
After that I merge upstream/main and push with --no-verify.

**Comment by twghu** [comment]
I'm reworking/testing for new PR

**Comment by htuch** [comment]
Suggest a force push to reset things here, thanks. Prefer not to do this generally, but it's the best option given the branch state.

**Comment by twghu** [comment]
New PR's added, same code, same corpus entries.  Added these PR's in case a cleaner push to upstream main is preferrred.

**Comment by phlax** [comment]
closing this in favour of the 2 breakout prs

**Pull request closed without merging**
